structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/controller/CDPStructuredEventV1Controller.java

```diff
@@ -29,17 +35,27 @@
     @Inject
     private CDPStructuredEventDBService structuredEventDBService;
 
+    @Inject
+    private CommonPermissionCheckingUtils permissionCheckingUtils;
+
     @Override
     @CustomPermissionCheck
     public List<CDPStructuredEvent> getAuditEvents(@ResourceCrn String resourceCrn, List<StructuredEventType> types, Integer page, Integer size) {
         PageRequest pageable = PageRequest.of(page, size, Sort.by("timestamp").descending());
-        return structuredEventDBService.getPagedEventsOfResource(types, resourceCrn, pageable).getContent();
+        permissionCheckingUtils.checkPermissionForUser(STRUCTURED_EVENTS_READ, ThreadBasedUserCrnProvider.getUserCrn());
+
+        List<CDPStructuredEvent> events = structuredEventDBService.getPagedEventsOfResource(types, resourceCrn, pageable).getContent();
+        checkPermissions(events);
+
+        return events;
     }
 
     @Override
     @CustomPermissionCheck
     public Response getAuditEventsZip(@ResourceCrn String resourceCrn, List<StructuredEventType> types) {
         Collection<CDPStructuredNotificationEvent> events = structuredEventDBService.getEventsOfResource(types, resourceCrn);
+        permissionCheckingUtils.checkPermissionForUser(STRUCTURED_EVENTS_READ, ThreadBasedUserCrnProvider.getUserCrn());
+        checkPermissions(events);
```

Comment:
so basically here you will check for the same right twice: once against the actual tenant, once against the given resource(s), is that what you want?